File: Data Structures & Algorithms/combination-target-sum/submission-0.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        sol = []
        nums.sort()

        def dfs(i, cur, cur_sum):
            if cur_sum == target:
                sol.append(cur.copy())
                return
            if cur_sum > target or i >= len(nums):
                return
            dfs(i + 1, cur.copy(), cur_sum)
            cur.append(nums[i])
            dfs(i, cur.copy(), cur_sum + nums[i])

        dfs(0, [], 0) 

        return sol

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        for (int i = 0, j = numbers.size() - 1; i < j;) {
            if (numbers[i] + numbers[j] == target) {
                return vector<int>{i+1, j+1};

            }
            else if ((numbers[i] + numbers[j]) < target) {
                i++;
            }
            else {
                j--;
            }
        }
        return std::vector<int>{0, 0};
    }
};
